scripts/commands/checkQiskitBotFiles.ts

```diff
@@ -0,0 +1,131 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2024.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import fs from "fs/promises";
+import path from "path";
+import { load } from "js-yaml";
+import { globby } from "globby";
+
+const ALLOWED_OWNERLESS_FILES = new Set([
+  "docs/start/_toc",
+  "docs/build/_toc",
+  "docs/build/bit-ordering",
+  "docs/build/operators-overview",
+  "docs/build/save-circuits",
+  "docs/transpile/_toc",
+  "docs/transpile/create-a-transpiler-plugin",
+  "docs/transpile/custom-backend",
+  "docs/transpile/transpiler-plugins",
+  "docs/verify/_toc",
+  "docs/verify/local-testing-mode",
+  "docs/verify/plot-quantum-states",
+  "docs/verify/using-ibm-quantum-simulators",
+  "docs/run/_toc",
+  "docs/run/circuit-execution",
+  "docs/run/configure-runtime-compilation",
+  "docs/run/execution-modes-faq",
+  "docs/run/execution-modes",
+  "docs/run/native-gates",
+  "docs/run/save-jobs",
+  "docs/run/visualize-results",
+]);
+
+const ALLOWED_INEXISTENT_FILES = new Set([
+  "docs/build/qubit-order",
+  "docs/build/operators_overview",
+  "docs/run/reserve-system-time",
+  "docs/migration-guides/index",
+  "docs/migration-guides/qiskit-runtime",
+  "docs/migration-guides/qiskit-runtime-examples",
+  "docs/migration-guides/qiskit-quantum-instance",
+  "docs/migration-guides/qiskit-algorithms-module",
+  "docs/migration-guides/qiskit-opflow-module",
+]);
+
+type qiskitBotConfig = {
+  always_notify: boolean;
+  notification_prelude: string;
+  notifications: { [id: string]: string[] };
+};
+
+const GLOBS = [
+  "docs/start/*",
+  "docs/build/*",
+  "docs/transpile/*",
+  "docs/verify/*",
+  "docs/run/*",
+  "docs/migration-guides/*",
+];
+
+async function main() {
+  const qiskitBotFiles = await getQiskitBotFiles();
+  const filesToCheck = await getFilesToCheck();
+
+  const missingFiles = [...filesToCheck].filter(
+    (file) => !qiskitBotFiles.has(file) && !ALLOWED_OWNERLESS_FILES.has(file),
+  );
```

Comment:
It'd be more readable if qiskitBotFiles returns an array so you don't have to do `[...]`. I'm not sure we need to be worried about duplicates with using a Set? 

It'd also be great to sort the file paths inside the helper functions so that the error message is easier to read.

Ditto `leftoverFiles`

Reply:
Probably not as we don't have a ton of files in the qiskit-bot config. We could have duplicated if we wrote the same file twice there but maybe it's not a big deal for this script. Having two messages of error in case of duplicated entries could even be helpful to detect them and remove them.